graders/c/cgrader/cgrader.py

```diff
@@ -459,7 +459,9 @@ def compile_re(t):
             comment = (
                 ""
                 if len(exp_output) == 1
-                else " one of" if must_match_all_outputs == "any" else " all of"
+                else " one of"
+                if must_match_all_outputs == "any"
+                else " all of"
```

Comment:
I actually find this whole thing kind of hard to read; could we split this up into consecutive `if`/`elif`? That said, let's maybe do this in a separate PR first. e.g.:

```py
if len(exp_putput) == 1:
    comment = ("")
elif ...
```

(Assuming I even got the first part of that right, which I'm not confident on, because this is so tricky to read lol)

Reply:
It might be worth adding type annotations/checking + doing a comprehensive pass through this file. There's a few places where the code is hard to read.

Reply:
This segment is rewritten in #9785 for clarity.

Reply:
Thanks @jonatanschroeder!